src/modules/dashboard/components/TaskComments/TaskComments.tsx

```diff
@@ -72,26 +76,27 @@ const handleKeyboardSubmit = (
   return false;
 };
 
-const TaskComments = ({
-  taskTitle,
-  colonyAddress,
-  draftId,
-  history,
-}: Props) => {
-  const { username, walletAddress } = useLoggedInUser();
-
-  const onSuccess: OnSuccess = useCallback(
-    (result, { resetForm, setStatus }) => {
-      setStatus({});
-      // @ts-ignore proper formik typings!
-      resetForm({ comment: '' });
-    },
-    [],
-  );
-
-  const transform = useCallback(
-    mergePayload({ colonyAddress, author: walletAddress, draftId, taskTitle }),
-    [colonyAddress, draftId],
+const TaskComments = ({ draftId, history }: Props) => {
+  const { username } = useLoggedInUser();
+
+  const [sendComment] = useSendTaskMessageMutation();
+  const onSubmit = useCallback(
+    ({ comment: message }: FormValues, { resetForm }: FormikBag<any, any>) =>
+      sendComment({
+        variables: {
+          input: {
+            id: draftId,
+            message,
+          },
+        },
+        refetchQueries: [
+          {
+            query: TaskFeedEventsDocument,
+            variables: { id: draftId, },
```

Comment:
The type of `variables` isn't inferred from the value of the `query` property, so we should probably use some type assertion here:

```suggestion
            variables: { id: draftId, } as TaskFeedEventsQueryVariables,
```